### PR TITLE
algorithms: add copy_if sender support

### PR DESCRIPTION
**Pull request by guptapratykshh**
## Proposed Changes

- Add sender support for `hpx::copy_if` when used with scheduler-backed task policies
- Preserve the existing non-scheduler execution path and return the correct output iterator from sender completion
- Add tests covering iterator categories, empty/all/none/mixed inputs, returned iterators, and exception propagation

### CODE REVIEW

**Comment by hkaiser** [comment]
@guptapratykshh Please have a look at the bot's comments to verify whether those are applicable and resolve them either by patching the code or adding a comment why you think it's ok not to address them.

File: libs/core/algorithms/include/hpx/parallel/algorithms/copy.hpp [hunk 1]
```diff
@@ -804,11 +803,27 @@ namespace hpx {
...
+
+            if constexpr (has_scheduler_executor)
+            {
+                return hpx::parallel::util::get_second_element(
+                    hpx::parallel::execution::async_execute(policy.executor(),
+                        [first, last, dest, pred = HPX_MOVE(pred)]() mutable {
+                            return hpx::parallel::detail::sequential_copy_if(
+                                first, last, dest, HPX_MOVE(pred),
+                                hpx::identity_v);
+                        }));
```
[the remark this answers is not in the document]
Comment (guptapratykshh) [reply libs/core/algorithms/include/hpx/parallel/algorithms/copy.hpp]: Addressed in 1f7871b: the scheduler path now routes exceptions through handle_local_exceptions, preserving hpx::exception_list for regular exceptions and direct std::bad_alloc propagation

File: libs/core/algorithms/tests/unit/algorithms/copy_if_sender.cpp [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        }
+        else
+        {
+            HPX_TEST(false);
+        }
+    }
+    catch (Exception const&)
+    {
+        caught_expected_exception = true;
+    }
```
[the remark this answers is not in the document]
Comment (guptapratykshh) [reply libs/core/algorithms/tests/unit/algorithms/copy_if_sender.cpp:178]: Addressed in 1f7871b: the test now rejects a directly propagated std::runtime_error and requires it through hpx::exception_list, while still accepting std::bad_alloc directly

File: libs/core/algorithms/include/hpx/parallel/algorithms/copy.hpp [hunk 3]
```diff
@@ -804,11 +805,36 @@ namespace hpx {
...
+                                        HPX_MOVE(pred), hpx::identity_v);
+                            }
+                            catch (...)
+                            {
+                                hpx::parallel::util::detail::
+                                    handle_local_exceptions<ExPolicy>::call(
+                                        std::current_exception());
+                            }
+                        }));
+            }
```
[the remark this answers is not in the document]
Comment (guptapratykshh) [reply libs/core/algorithms/include/hpx/parallel/algorithms/copy.hpp:834]: Addressed in 86f11d0: the scheduler result now goes through algorithm_result::get, preserving an iterator for non-task policies and a sender for task policies, with compile-time and runtime coverage

**Comment by hkaiser** [comment]
Please rebase this onto master, that will fix some of the remaining CI problems.
